libraries/adaptive-expressions/tests/expressionParser.test.js

```diff
@@ -464,7 +464,7 @@ const dataSource = [
     ['date(timestamp)', '3/15/2018'],//Default. TODO
     ['year(timestamp)', 2018],
     ['length(utcNow())', 24],
-    ['utcNow(\'MM-DD-YY HH\')', moment(new Date().toISOString()).utc().format('MM-DD-YY HH')],
+    ['utcNow(\'MM-DD-YY HH\')', 'getNowTime'],
```

Comment:
Is this a case of "two new date objects that are supposed to have the same time but since they were instantiated at different times the test intermittently fails?"

Reply:
Because the tests in this file take a long time to run?

Reply:
I'm still not 100% on this, but:

> Is this a case of "two new date objects that are supposed to have the same time but since they were instantiated at different times the test intermittently fails?"

I believe so.

> Because the tests in this file take a long time to run?

[This is my pipeline test that broke](https://fuselabs.visualstudio.com/SDK_v4/_build/results?buildId=150473&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=d38544d5-e3b6-5628-9a8d-57f5f7ecd74f&l=3408). If you look at the [raw output](https://fuselabs.visualstudio.com/86659c66-c9df-418a-a371-7de7aed35064/_apis/build/builds/150473/logs/29), all tests began at 16:58, but this particular test file doesn't get hit until 17:01. So, I _think_ that mocha runs all of the test side effects before actually running the tests.